middleware/kubernetes/handler.go

```diff
@@ -53,38 +49,46 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 			if (dns.CountLabel(name) - 1) < k.AutoPath.NDots {
 				break
 			}
-			// Search "svc.cluster.local" and "cluster.local"
+			origQName := state.QName()
+			// Search "svc.cluster.local." and "cluster.local."
 			for i := 0; i < 2; i++ {
```

Comment:
it strips the first segment from path each time through.
The original path is ... `w.x.y.z`, then the first loop does `x.y.z`, and the second does `y.z`